dashboard/test/jobs/application_job_test.rb

```diff
@@ -11,29 +11,16 @@ def perform
     end
   end
 
-  test 'jobs log JobCount when enqueued' do
-    # When testing, jobs aren't queued to the DB, so mock that.
-    Delayed::Job.stubs(:count).returns(1)
-
-    Cdo::Metrics.expects(:push).with(
-      ApplicationJob::METRICS_NAMESPACE,
-      all_of(
-        includes_metrics(JobCount: 1),
-        includes_dimensions(:JobCount, Environment: CDO.rack_env)
-      )
-    )
-
-    TestableJob.perform_later
-  end
-
-  test 'enqueued jobs log JobCount, WaitTime, ExecutionTime, and TotalTime' do
-    # after_enqueue metrics
-    Delayed::Job.stubs(:count).returns(1)
+  test 'enqueued jobs log PendingJobCount FailedJobCount WaitTime ExecutionTime and TotalTime' do
+    # TODO: actually assert the expected values for job counts
+    # When implementing, I ran into a lot of difficulty mocking
+    # the Delayed::Job.where(failed_at: nil) calls accurately.
```

Comment:
I don't recall ever having to mock these kinds of calls -- rather I have always relied on the database transaction to isolate my test case, and then done something like this: https://github.com/code-dot-org/code-dot-org/blob/43d2f4652e17d6919979541e8ad54438db0e3413/dashboard/test/controllers/programming_classes_controller_test.rb#L180-L183

Reply:
I believe that `perform_later` during tests does not use the database, but maybe it should?

Reply:
hmm good point, I mistook this for something resembling active record.

now that you mention it, I believe we are not even using the delayed job adapter during unit tests, rather we are using:
* https://api.rubyonrails.org/classes/ActiveJob/QueueAdapters/TestAdapter.html
* https://api.rubyonrails.org/classes/ActiveJob/TestHelper.html

It looks like the test adapter has `ActiveJob::QueueAdapters::TestAdapter.enqueued_jobs`, in case that's helpful?